Shared/ViewModels/APIKeyViewModel.swift

```diff
@@ -0,0 +1,179 @@
+//
+// Swiftfin is subject to the terms of the Mozilla Public
+// License, v2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at https://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2024 Jellyfin & Jellyfin Contributors
+//
+
+import Combine
+import Foundation
+import JellyfinAPI
+import OrderedCollections
+
+final class APIKeyViewModel: ViewModel, Eventful, Stateful {
+
+    // MARK: Event
+
+    enum Event {
+        case error(JellyfinAPIError)
+        case created
+        case deleted
+        case content
+    }
+
+    // MARK: Action
+
+    enum Action: Equatable {
+        case getAPIKeys
+        case createAPIKey(name: String)
+        case deleteAPIKey(key: String)
+    }
+
+    // MARK: State
+
+    enum State: Hashable {
+        case initial
+        case error(JellyfinAPIError)
+        case content
+    }
+
+    // MARK: Event Publishing
+
+    var events: AnyPublisher<Event, Never> {
+        eventSubject
+            .receive(on: RunLoop.main)
+            .eraseToAnyPublisher()
+    }
+
+    // MARK: Event Publishing
+
+    private var AuthenticationTokenTask: AnyCancellable?
+    private var eventSubject: PassthroughSubject<Event, Never> = .init()
+
+    @Published
+    final var backgroundStates: OrderedSet<BackgroundState> = []
+    @Published
+    final var apiKeys: OrderedDictionary<String, BindingBox<AuthenticationInfo?>> = [:]
```

Comment:
Just for future reference could you explain why some things are in a box and others aren't? I was basing this on how ActiveSessions came out.

If I had to guess, is it the updating? Like, the box, because it's indexed, lets us add/remove items more efficiently than an array?

Reply:
Sessions are in a box because they are being updated from another source that updates frequently and we allow pushing to the session details. The session is still updating while in that detail view so we have the box update the session. Now, it is a bit over engineered and I have realized that but didn't change it back to just have the detail view have its own view model filter sessions based on the device ID.

https://github.com/jellyfin/Swiftfin/pull/1230#issuecomment-2384465881

These are pretty static and don't require polling for more information, we don't need to propagate that information elsewhere so a box isn't needed.

I don't know why I missed this on the devices PR, but those don't need to be in a `BindingBox` either as they are also static.

Reply:
> I don't know why I missed this on the devices PR, but those don't need to be in a BindingBox either as they are also static.

Gotcha! Sorry that I missed that from the original active sessions. I'd be happy to make another PR to fix up in devices?

The other one I used a box for was users, but that one has an observer. So, I think the box is correct for that one?

Reply:
No, we also don't need boxes for that.